app/src/main/java/com/example/househomey/HomeSelectStateFragment.java

```diff
@@ -0,0 +1,138 @@
+
+        itemListView = rootView.findViewById(R.id.item_list);
+        itemAdapter = new ItemAdapter(getContext(), itemList);
+        itemListView.setAdapter(itemAdapter);
+        itemView = (ItemAdapter) itemAdapter;
+        itemView.setSelectState(true);
+        itemRef.addSnapshotListener(this::setupItemListener);
+
+        final Button cancelButton = rootView.findViewById(R.id.cancel_select_button);
+        cancelButton.setOnClickListener(v -> {
+            unselectAllItems();
+            navigateToFragmentPage(getContext(), new HomeBaseStateFragment(itemRef));
+        });
+
+        final Button deleteButton = rootView.findViewById(R.id.action_delete);
+        deleteButton.setOnClickListener(v -> {
+            ArrayList<Item> selectedItems = getSelectedItems();
+            if (selectedItems.size()>0) {
+                DeleteItemsFragment fragment = new DeleteItemsFragment(this, selectedItems);
+                fragment.show(requireActivity().getSupportFragmentManager(),"Delete Items");
+            }
+            else {
+                Toast.makeText(requireActivity().getApplicationContext(),
+                        "Please select one or more items to delete.",
+                        Toast.LENGTH_SHORT).show();
+            }
+        });
+
+        return rootView;
+    }
+
+    /**
+     * This is an interface method that handles the deletion of selected items from the user database
+     * This is done upon user confirmation in the delete items dialog
+     * @see DeleteItemsFragment
+     * @param selectedItems Items to be deleted
+     */
+    @Override
+    public void onOKPressed(ArrayList<Item> selectedItems){
+        WriteBatch batch = FirebaseFirestore.getInstance().batch();
+        for (Item item : selectedItems) {
+            batch.delete(itemRef.document(item.getId()));
+        }
+        batch.commit()
+                .addOnSuccessListener((result) -> {
+                    Log.i("Firestore", "Items successfully deleted");
+                })
+                .addOnFailureListener((error) -> {
+                    Log.e("Firestore", "Failed to remove items.", error);
+                });
+    }
+
+    /**
+     * This an interface method that sets the title of the Delete items dialog
+     * It asks us the user to confirm the deletion of the selected items
+     * @see DeleteItemsFragment
+     * @param selectedItems items selected to be deleted
+     * @return string asking confirmation of deletion #selectedItems
+     */
+    @Override
+    public String DialogTitle(ArrayList<Item> selectedItems) {
+        return "Confirm deletion of "+selectedItems.size()+" item(s)?";
+    }
+
+    /**
+     * This method checks for all the selected items in the list and returns them
+     * @return List of selected items
+     */
+    public ArrayList<Item> getSelectedItems() {
+        ArrayList<Item> selectedItems = new ArrayList<>();
+        for (Item item : itemList) {
+            if (item.isSelected()) {
```

Comment:
My main comment here would be that `isSelected` isn't a property of our Item object. It's a property of the list. So we probably don't want to store the selected state inside the Item class.

Instead, I think it's possible to loop through the views of the ArrayAdapter to get selected state right from the checkboxes (so you wouldn't even need a checkbox listener). See below:
```java
ArrayList<Item> selectedItems = new ArrayList<>();
        for (int i = 0; i < itemList.size(); i++) {
            View view = itemAdapter.getView(0, null, null);
            CheckBox checkbox = view.findViewById(R.id.item_checkBox);
            if (checkbox.isSelected()) {
                selectedItems.add(itemList.get(i));
            }
        }
```
If it's possible to use this approach throughout the code instead of via Item, I think it encapsulates the info better

Reply:
I think there was some reason as to why I decided to have isSelected be a property of the Item class, but I can't remember it now lol. I think it had something to do with if we wanted to select items for some other functionality that didn't involve selecting them by checking the checkboxes. 

But yeah, I don't see a reason why we can't just have the items be selected through the ItemAdapter only. Will make the change.